internal/ast/ast.go

```diff
@@ -434,24 +434,12 @@ func IsLValue(expr Expr) bool {
 // Stmt is the abstract syntax tree for any AWK statement.
 type Stmt interface {
 	Node
+	StartPos() Position // position of first character belonging to the node
+	EndPos() Position   // position of first character immediately after the node
```

Comment:
Yep, nice. But let's call them just `Start()` and `End()`. I don't think "pos" adds anything here, as the words "start" and "end" already convey position, and `Position` is the return type. In addition, `go/types` uses `Start()` and `End()`, setting something of a precedent for ASTs.

Reply:
Well, `Start`/`End` clashes with the relevant fields in `*Stmt` structs.

Reply:
Yeah, that's fine, just make them non-exported fields `start` and `end`.

Reply:
Unfortunately this prevents structs instantiation 
![image](https://user-images.githubusercontent.com/11706893/196034662-7fb08fce-54d0-4c0b-8ede-ce3a86c14377.png)

Reply:
Ah, of course. Okay, `StartPos` / `EndPos` is fine for now then.